services/blockchain-app-registry/shared/metadata.js

```diff
@@ -248,10 +248,10 @@ const getBlockchainAppsTokenMetadata = async (params) => {
 		const { tokenName, ...remParams } = params;
 		params = remParams;
 
-		// chainID or chainName must be specified with the network
+		// chainName must be specified with the network
 		// Skip when tokenID is specified, network can be resolved automatically
 		if (!('tokenID' in params) && !('chainID' in params) && ((!('chainName' in params)) || (!('network' in params)))) {
-			throw new InvalidParamsException('\'tokenName\' must be specified with either \'chainID\', or \'chainName\' and \'network\'.');
+			throw new InvalidParamsException('\'tokenName\' must be specified with either \'chainID\', or \'tokenID\', or (\'chainName\' and \'network\').');
```

Comment:
I think we can completely remove this condition. We added this condition initially when the network was supposed to be mandatory parameter which is not the case now. It works fine after removing the given condition as well. What do you say @sameersubudhi?

Reply:
Yes, let's remove this check. The end-users can filter the results as per requirements. Also, please ensure that the docs are up-to-date in this regard.

Reply:
Applied.